search/elastic.py

```diff
@@ -78,197 +98,178 @@ def _get_filter_field(field_name, field_value):
         filter_field = {
             "terms": {
                 field_name: field_value
-            }
-        }
-    else:
-        filter_field = {
-            "term": {
-                field_name: field_value
-            }
+            },
         }
     return filter_field
 
 
 def _process_field_queries(field_dictionary):
     """
-    We have a field_dictionary - we want to match the values for an elasticsearch "match" query
-    This is only potentially useful when trying to tune certain search operations
+    We have a field_dictionary - we match the values using a "term" filter
+    in elasticsearch
     """
-    def field_item(field):
-        """ format field match as "match" item for elasticsearch query """
-        return {
-            "match": {
-                field: field_dictionary[field]
-            }
-        }
-
-    return [field_item(field) for field in field_dictionary]
-
-
-def _process_field_filters(field_dictionary):
-    """
-    We have a field_dictionary - we match the values using a "term" filter in elasticsearch
-    """
-    return [_get_filter_field(field, field_value) for field, field_value in field_dictionary.items()]
+    return [
+        _get_filter_field(field, field_value)
+        for field, field_value in field_dictionary.items()
+    ]
 
 
 def _process_filters(filter_dictionary):
     """
-    We have a filter_dictionary - this means that if the field is included
-    and matches, then we can include, OR if the field is undefined, then we
-    assume it is safe to include
+    Method which build list for filtering, adds expression
+    if field doesn't exists in document.
     """
-    def filter_item(field):
-        """ format elasticsearch filter to pass if value matches OR field is not included """
-        if filter_dictionary[field] is not None:
-            return {
-                "or": [
-                    _get_filter_field(field, filter_dictionary[field]),
-                    {
-                        "missing": {
-                            "field": field
-                        }
-                    }
-                ]
-            }
-
-        return {
-            "missing": {
-                "field": field
-            }
-        }
-
-    return [filter_item(field) for field in filter_dictionary]
+    filter_expression = []
+    for field in filter_dictionary:
+        if filter_dictionary[field]:
+            filter_expression.extend([
+                _get_filter_field(field, filter_dictionary[field])
+            ])
+        filter_expression.extend([
+            {
+                "bool": {
+                    "must_not": {"exists": {"field": field}, },
+                },
+            },
+        ])
+    return filter_expression
```

Comment:
I think that this code is cleaner without using generator. Before I removed method which made the same action but it was big

Reply:
Hmm, I think that we did not understand each other. You definitely wrote about the generator?

Just see how it could be
```
def _process_filters(filter_dictionary):
    """
    Method which build list for filtering, adds expression
    if field doesn't exists in document.
    """
    for field in filter_dictionary:
        if filter_dictionary[field]:
            yield _get_filter_field(field, filter_dictionary[field])
        yield {
                "bool": {
                    "must_not": {"exists": {"field": field}, },
                },
            }
```
It is easier and simpler than what you had. Plus, the generator has many advantages. I think you know about them.

Reply:
Ok, done